TopMovies/Core/State/MoviesListState.swift

```diff
@@ -10,25 +10,30 @@ import ReSwift
 typealias MoviesRelational = [Movie.ID: Movie]
 
 struct MoviesListState: StateType {
-     let moviesList: RequestState<[Movie.ID]>
-     let relational: MoviesRelational 
+    let moviesListName: String
+    let moviesList: RequestState<[Movie.ID]>
```

Comment:
If you would like to support categories, you should think for the future and expect that there will be more categories and that they may be dynamic.
So here's an example of how you could describe it.

```swift
struct MovieCategory {
 //let id, let title...
}

struct MovieCategoriesState: StateType {
  let relational: [MovieCategory.ID: MovieCategory]
  let categoriesList: RequestState<[MovieCategory.ID]>
  let categoryRequests: [MovieCategory.ID: RequestState<[Movie.ID]>] 
  // after you fetch "all" categories, you can store each category as "state.requests[categoryId] = event.categories[id]?.map(\.id).map(RequestState.completed) ?? nil" 
}

struct MoviesState: StateType {
  let relational: MoviesRelational
}
```

With a similar approach you will be able to:
a) describe all categories fetching in the state using single RequestState.
b) store Category in state as a strong type.
c) perform reload or load more request only for one specific category.

Feel free to design that structure as you wish to be able to handle those two cases.